source/frontend/src/components/common/mapFSM/MapStateMachineContext.tsx

```diff
@@ -0,0 +1,275 @@
+  isSidebarOpen: boolean;
+  pendingFlyTo: boolean;
+  requestedFlyTo: RequestedFlyTo;
+  mapFeatureSelected: FeatureSelected | null;
+  mapLocationSelected: LatLngLiteral | null;
+  mapLocationFeatureDataset: LocationFeatureDataset | null;
+  selectedFeatureDataset: LocationFeatureDataset | null;
+  showPopup: boolean;
+  isLoading: boolean;
+  mapFilter: IPropertyFilter | null;
+  mapFeatureData: MapFeatureData;
+  draftLocations: LatLngLiteral[];
+  pendingFitBounds: boolean;
+  requestedFitBounds: LatLngBounds;
+  iSelecting: boolean;
+
+  requestFlyToLocation: (latlng: LatLngLiteral) => void;
+  requestFlyToBounds: (bounds: LatLngBounds) => void;
+  processFlyTo: () => void;
+  processFitBounds: () => void;
+  openSidebar: (sidebarType: SideBarType) => void;
+  closeSidebar: () => void;
+  closePopup: () => void;
+
+  mapClick: (latlng: LatLngLiteral) => void;
+  mapMarkerClick: (featureSelected: FeatureSelected) => void;
+
+  setMapFilter: (mapFilter: IPropertyFilter) => void;
+  refreshMapProperties: () => void;
+  prepareForCreation: () => void;
+  startSelection: () => void;
+  finishSelection: () => void;
+  setDraftLocations: (locations: LatLngLiteral[]) => void;
+}
+
+const MapStateMachineContext = React.createContext<IMapStateMachineContext>(
+  {} as IMapStateMachineContext,
+);
+
+export function useMapStateMachine() {
+  const context = React.useContext(MapStateMachineContext);
+  if (context === undefined) {
+    throw new Error('useMapStateMachine must be used within a MapStateMachineContextProvider');
+  }
+  return context;
+}
+
+export const MapStateMachineProvider: React.FC<React.PropsWithChildren<unknown>> = ({
+  children,
+}) => {
+  const locationLoader = useLoactionFeatureLoader();
+  const mapSearch = useMapSearch();
+  const history = useHistory();
+
+  const service = useInterpret(mapMachine, {
+    actions: {
+      navigateToProperty: (context, event: any) => {
+        const selectedFeatureData = context.mapLocationFeatureDataset;
+        if (selectedFeatureData?.pimsFeature?.properties.PROPERTY_ID) {
+          const pimsFeature = selectedFeatureData.pimsFeature;
+          history.push(`/mapview/sidebar/property/${pimsFeature.properties.PROPERTY_ID}`);
+        } else if (selectedFeatureData?.parcelFeature?.properties.PID) {
+          const parcelFeature = selectedFeatureData.parcelFeature;
+          const parsedPid = pidParser(parcelFeature.properties.PID);
+          history.push(`/mapview/sidebar/non-inventory-property/${parsedPid}`);
+        }
+      },
+    },
+    services: {
+      loadLocationData: (context, event: any) => {
+        let latLng: LatLngLiteral = { lat: 0, lng: 0 };
+        if (event.type === 'MAP_CLICK') {
+          latLng = event.latlng;
+        } else if (event.type === 'MAP_MARKER_CLICK') {
+          latLng = event.featureSelected.latlng;
+        }
+        const result = locationLoader.showLocationDetails(latLng, context.isSelecting);
```

Comment:
should else throw an exception instead of showing the location details at 0,0?

Reply:
I think this will be solved once the state machine gets strongly typed.